tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,486 @@
+        ] = None,
+    ) -> None:
+        super().__init__(
+            name=name,
+            evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+            data_context=None,
+        )
+
+        self._call_count: int = 0
+
+    @property
+    def call_count(self) -> int:
+        return self._call_count
+
+    def _build_parameters(
+        self,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> Attributes:
+        """
+        Only "_call_count" is incremented; otherwise, arbitrary test value is returned.
+        """
+        self._call_count += 1
+
+        return Attributes(
+            {
+                FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY: 1,
+                FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY: {"batch_id_0": 1},
+                FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY: {"environment": "test"},
+            }
+        )
+
+
+# noinspection PyUnresolvedReferences,PyUnusedLocal
+@pytest.mark.unit
+@mock.patch(
+    "great_expectations.rule_based_profiler.parameter_builder.parameter_builder.convert_to_json_serializable",
+    return_value="my_json_string",
+)
+def test_parameter_builder_should_not_recompute_evaluation_parameter_builders_if_precomputed(
+    mock_convert_to_json_serializable: mock.MagicMock,
+):
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=None,
+    )
+
+    domain = DummyDomain()
+
+    parameter_container = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    my_evaluation_dependency_0_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_0_parameter_builder.call_count == 1
+
+    my_evaluation_dependency_1_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_1_parameter_builder.call_count == 1
+
+    dependencies_fully_qualified_parameter_names: List[
```

Comment:
what can we assert here? Why is this setup step needed?

Specifically, I should know what `get_fully_qualified_parameter_names` is expected to return given this setup, and create a specific unit test for this method.

Reply:
@donaldheppner As I mentioned in the introductory docstring, `get_fully_qualified_parameter_names()` has been thoroughly unit tested (in "tests/rule_based_profiler/parameter_builder/test_parameter_container.py").  Here, it is used as a fast utility so that the actual contents of the shared memory are tracked truthfully as the ParameterBuilders are executed.  We can assert on what these are:
```
    assert dependencies_fully_qualified_parameter_names == [
        "$parameter.raw.my_evaluation_dependency_parameter_name_1",
        "$parameter.raw.my_evaluation_dependency_parameter_name_0",
        "$parameter.my_evaluation_dependency_parameter_name_1",
        "$parameter.my_evaluation_dependency_parameter_name_0",
    ]
```
but I think that we do not want to put this assertion into this test module, because it has already been done elsewhere, as I mentioned, so it would be redundant; instead, we use this fast utility to track the contents of the shared memory.  

Please notice that when we do have a different situation in the next test (`test_parameter_builder_dependencies_evaluated_in_parameter_builder_if_not_precomputed()`), where we do test exactly for the resolution (lines 303 and 327) -- and then we must assert against the known expected list of names.

Please let me know if this clarifies the situation and/or if you have further suggestions.  Thank you.